ui/app/AppLayouts/ActivityCenter/controls/NotificationAvatar.qml

```diff
@@ -0,0 +1,186 @@
+    property real  density: 1.0
+
+    // Base sizes (will be multiplied by density)
+    property int   baseAvatarSize: 36
+    property int   baseBadgeSize:  18
+
+    /// Badge background color
+    property color badgeBgColor: Theme.palette.primaryColor3
+
+    // How much the badge overlaps the avatar on bottom-right (0..1)
+    property real  badgeOverlapRatio: 0.25
+
+    // If true, root.implicitWidth/Height include the badge extents.
+    // If false, the implicit size hugs only the avatar.
+    property bool  includeBadgeInImplicit: true
+
+    // When true (default), the avatar area reacts visually to mouse hover and click
+    // and exposes a pointing-hand cursor. Disable if you don’t want the
+    // avatar to be an interactive target.
+    property bool isAvatarClickable: true
+
+    // When true (default), the badge overlay reacts visually to mouse hover and click
+    // and exposes a pointing-hand cursor. Disable if you don’t want the
+    // badge to be clickable or highlight on hover.
+    property bool isBadgeClickable: true
+
+    // Interactions
+    signal avatarClicked()
+    signal badgeClicked()
+
+    QtObject {
+        id: d
+
+        // Dynamic size properties
+        readonly property int avatarSize: Math.round(baseAvatarSize * density)
+        readonly property int badgeSize:  Math.round(baseBadgeSize  * density)
+    }
+
+    // ──────────────────────────────────────────────────────────────────────────
+    // Layout sizing
+    // ──────────────────────────────────────────────────────────────────────────
+    implicitWidth:  root.includeBadgeInImplicit
+                    ? Math.max(avatarImg.width,  badge.x + badge.width)
+                    : avatarImg.width
+    implicitHeight: root.includeBadgeInImplicit
+                    ? Math.max(avatarImg.height, badge.y + badge.height)
+                    : avatarImg.height
+
+    // ──────────────────────────────────────────────────────────────────────────
+    // Content image (single source for both masked and unmasked paths)
+    // ──────────────────────────────────────────────────────────────────────────
+    Image {
+        id: avatarImg
+        anchors.top: parent.top
+        anchors.left: parent.left
+        width: d.avatarSize
+        height: width
+        source: root.avatarSource
+        fillMode: Image.PreserveAspectCrop
+        smooth: true
+        mipmap: true
+
+        layer.enabled: root.circular
+        layer.effect: MultiEffect {
+            source: avatarImg
+
+            maskEnabled: root.circular
+            maskSource: circleMask
+
+            visible: root.circular
+            enabled: root.circular
+
+            maskThresholdMin: 0.5
+            maskSpreadAtMin: 1.0
+        }
+
+        // Mask geometry
+        Rectangle {
+            id: circleMask
+            anchors.fill: avatarImg
+            radius: width / 2
+            visible: false
+            layer.enabled: true
```

Comment:
Why is the `layer` enabled here? Really needed for the effect?

Reply:
If not, no image shown!